ocaml/typing/ctype.ml

```diff
@@ -3844,7 +3849,7 @@ let filter_arrow env t l ~force_tpoly =
       constrain_type_layout_exn env Unify t' layout;
       arrow_desc
   | Tarrow((l', arg_mode, ret_mode), ty_arg, ty_ret, _) ->
-      if l = l' || !Clflags.classic && l = Nolabel && not (is_optional l')
+      if l = l' || !Clflags.classic && l = Nolabel && not (is_omittable l')
```

Comment:
I would still use `equivalent_in_classic_mode` here, not `is_omittable`. Yes, it's a little duplicative, but it's clearer to readers.

Reply:
This accepts a strict subset of cases accepted by `equivalent_in_classic_mode`; if `l=Labelled x` and `l'=Nolabel y` then the check fails (or if they're both labelled, but with different names). I did want to ask why this case is different from the others though. 
Intuitively, I see that `l'` is the expected label, and trying to match that up with a function that doesn't satisfy either A. same label or B. is unlabelled when the expected is labelled doesn't quite work. But I don't completely understand why we care about this restriction here when we don't in other places.

Reply:
Oh -- I was thinking to have `l = Nolabel && equivalent_in_classic_mode l l'`, which should accept the same situations